### ISSUE TITLE
Referrer should not equal url

### ISSUE THREAD

**Issue by wheresrhys** [Financial-Times/spoor-enrichment#172]
We have a lot of data where `page.referrer === page.url`, which muddies what referrer is supposed to mean. There's a line in here which sets url to referrer if `context.url` is undefined. Not exactly sure which events that's there to cover. If it does need to be there it could do with tweaking to remove the ambiguity from the data.

**Comment by commuterjoy** [comment]
Lucas Gluchowski [1:35 PM] 
for internal referrer usually the article itself is the top referrer which is a bit odd. I thought it's the case when user refresh the page, hence the referrer is the current page. I played a bit and refreshed and article but the page views have no referrer so that's not the case. What is triggering self-referring?

​[1:35] 
BTW I know it's Friday, I shouldn't ask difficult questions :simple_smile:

Matt Chadburn [1:37 PM] 
https://github.com/Financial-Times/spoor-enrichment/issues/172 - yes, it's been mentioned

​[1:38] 
https://github.com/Financial-Times/spoor-enrichment/blob/master/server/transforms/url.js#L27

​[1:39] 
there's some confusing logic where we pick the location up from the referer (iirc because tracking pixels will report the referer as the current location)

​[1:39] 
but it's quite oblique, so I was thinking of taking it out

**Comment by markstephens** [comment]
yes, I don't think you can work our referrer from enrichment

This line should just look for the referrer value within the context object?
https://github.com/Financial-Times/spoor-enrichment/blob/master/server/transforms/url.js#L26

**Comment by adambraimbridge** [comment]
It looks like https://github.com/Financial-Times/spoor-enrichment/blob/master/server/transforms/url.js#L50 is the culprit. I suggest changing it so that instead of setting the url to the referer, set it to null.

**Comment by markstephens** [comment]
That line isn't ideal, but it's better than having null.
That line uses spoor-api's referrer, not the user's referrer.

The original suggestion in this issue that url shouldn't equal referrer isn't quite right as there are obviously many reasons why that would happen in the real world.

**Comment by adambraimbridge** [comment]
Yeah, looks like this issue could bear further cognition. @markstephens, let's discuss it next week.
